lib/ecto/adapters/mysql/connection.ex

```diff
@@ -432,35 +432,28 @@ if Code.ensure_loaded?(Mariaex.Connection) do
     alias Ecto.Migration.Index
     alias Ecto.Migration.Reference
 
-    def ddl_exists(%Table{name: name}) do
-      """
-      SELECT COUNT(1)
-        FROM information_schema.tables
-       WHERE table_schema = SCHEMA()
-             AND table_name = '#{escape_string(to_string(name))}'
-      """
-    end
+    def execute_ddl({:create, %Table{} = table, columns}) do
+      engine = engine_expr(table.engine)
+      options = options_expr(table.options)
 
-    def ddl_exists(%Index{name: name}) do
-      """
-      SELECT COUNT(1)
-        FROM information_schema.statistics
-       WHERE table_schema = SCHEMA()
-         AND index_name = '#{escape_string(to_string(name))}'
-      """
+      "CREATE TABLE #{quote_table(table.name)} (#{column_definitions(columns)})" <> engine <> options
```

Comment:
We can share the logic between `:create` and `:create_if_not_exists` and avoid the duplication here.